profiles/logical/holo/holo-business-services/default.nix

```
@@ -0,0 +1,173 @@
+# This is a holo profile for holo business services
+# This set's up node js api's created inside holo-business-services
+# These services should be setup in systemd and you can use 
+# systemctl command to start or stop them
+#
+# e.g.
+#     systemctl start hbs-[service-name].service
+#     systemctl start hbs-registration.service
+
+{ config, lib, pkgs, ... }:
+
+with pkgs;
+
+let
+  registrationSocket = "unix:/run/hbs-registration/hbs-registration.sock";
+
+  authSocket = "unix:/run/hbs-auth/hbs-auth.sock";
+  
+  holochainWorkingDir = "/var/lib/holochain-rsm";
+
+  configureHolochainWorkingDir = "/var/lib/configure-holochain";
+
+  settings = import ../../global-settings.nix;
+
+  networks = import ../../holo-networks.nix;
+
+  holoNetwork = networks.selectNetwork config.system.holoNetwork;
+
+  mongoUri = (builtins.readFile /var/lib/membrane-proof-service/mongodb-uri);
+in
+
+{
+  imports = [
+    ../.
+    ../logging.nix
+  ];
+  environment.systemPackages = with holochainAllBinariesWithDeps.hpos; [ hc-state git hpos-update-cli hpos-holochain-client holochain hc kitsune-p2p-tx2-proxy ];
+
+  networking.firewall.allowedTCPPorts = [ 80 443 ];
+
+  security.acme = {
+    acceptTerms = true;
+    # REVIEW: maybe a dedicated email for Hydra?
+    email = "przemek.klimek@holo.host";
+  };
+
+  services.lair-keystore = {
+    enable = true;
+    # To not expect lair to preload a seed into the keystore, set preStart to false
+    preStart = "false";
+    default-password = settings.DEVICE_SEED_DEFAULT_PASSWORD;
+  };
+
+  services.hbs-registration = {
+    enable = true;
+    socket = registrationSocket;
+    configureHolochainConfig = "${configureHolochainWorkingDir}/config.yaml";
+    mongoUri = mongoUri;
+  };
+
+  services.hbs-auth = {
+    enable = true;
+    mongoUri = mongoUri;
+    socket = authSocket;
+  };
+
+  services.nginx = {
+    enable = true;
+
+    virtualHosts.publicApi = {
+      enableACME = true;
+      forceSSL = true;
+      locations = {
+        "/registration" = {
+          proxyPass = "http://${registrationSocket}:/";
+        };
+
+        "/auth" = {
+          proxyPass = "http://${authSocket}:/";
+        };
+
+        "/api/v1/ws/" = {
+          proxyPass = "http://127.0.0.1:42233";
+          proxyWebsockets = true;
+        };
+      };
+      serverName = holoNetwork.holoBusinessServicesServer.dns;
+    };
+
+    virtualHosts.localhost = {
+      locations."/holochain-api/".proxyPass = "http://unix:/run/hpos-holochain-api/hpos-holochain-api.sock:/";
+      locations."/holo-business-services/registration/".proxyPass = "http://${registrationSocket}:/";
+      locations."/holo-business-services/auth/".proxyPass = "http://${authSocket}:/";
```

```suggestion
      locations."/holochain-api/".proxyPass = "http://unix:/run/hpos-holochain-api/hpos-holochain-api.sock:/";
```
`holochain-api` needs to be available at http://localhost only for tests. Other two services are not used in NixOs tests, so those two lines are not required